### ISSUE TITLE
Uncaught TypeError: Cannot read properties of undefined (reading 'childNodes')

### ISSUE THREAD

**Issue by ChristianThingKnudsen** [DevExpress/testcafe#7489]
### What is your Scenario?

When running running an Elm application using the testcafe browser it causes an elm runtime error:
`Uncaught TypeError: Cannot read properties of undefined (reading 'childNodes')`
This is due to testcafe somehow manipulating the DOM that makes Elm chrash. 

### What is the Current behavior?

The Elm application chrashes

### What is the Expected behavior?

The Elm application should not chrash

### What is your public website URL? (or attach your complete example)

Unable to provide due to company restrictions

### What is your TestCafe test code?

```
fixture("Elm Fixture").page("url-to-elm-application");

test("Should render elm page", async (t) => {
  await elmPage.isRendered()
});
```

### Screenshots

![SCR-20230202-ddn](https://user-images.githubusercontent.com/27851699/216278834-150bb990-4b81-423f-84ba-ceb9b980df8f.png)

### Steps to Reproduce

1. Go to page of elm application.
2. The page is not rendered 
3. The console is spammed with errors

### TestCafe version

v1.20.1

### Node.js version

v1.20.1

### Command-line arguments

./node_modules/.bin/testcafe --skip-js-errors Chrome <./path-to-fixture>

### Browser name(s) and version(s)

Chrome 

### Platform(s) and version(s)

MacOS 12.16.1

### Other

We only see this issue on some elm pages. However the elm application runs perfectly outside TestCafe in Chrome, Firefox and Safari. I have tried using many different versions of testcafe but this seems to be an issue in all versions.

**Comment by ChristianThingKnudsen** [comment]
Unfortunately, I am unable to provide an example in a public forum due to company restrictions. Is it possible to provide an example for you to debug without uploading it to a public forum?

**Comment by AlexKamaev** [comment]
You can send a simple sample to support@devexpress.com.

**Comment by miherlosev** [comment]
Hi @ChristianThingKnudsen

Could you please upgrade to the latest version (`testcafe@.2.3.1`), specify the [dev](https://testcafe.io/documentation/402639/reference/command-line-interface#--dev) option, run the test and share a screenshot of the script error (the same as in the issue description)?

**Comment by ChristianThingKnudsen** [comment]
Upgrading to testcafe version `2.3.1` did not solve the issue. The same error persists. I use a custom runner activated by an npm script:

```
"test": "npm run type-check && ./node_modules/.bin/testcafe --skip-js-errors chrome $*"
```

And I provide the following config for my runner:

```
export const runnerConfig: Partial<RunOptions> = {
  skipJsErrors: true, // Should ideally be set to false
  quarantineMode: { successThreshold: 1, attemptLimit: 3 }, // Tests are only marked as failed when they fail three times
  selectorTimeout: 20000,
  assertionTimeout: 20000,
  pageLoadTimeout: 20000,
  speed: 1,
  browserInitTimeout: 720000,
};
```

In my case, the page does not load due to the above error and I get hundreds of console errors:

```
Uncaught TypeError: Cannot read properties of undefined (reading 'childNodes')
```

As mentioned, the above code solves it, but it removes the input field.

**Comment by miherlosev** [comment]
You need to update the `Runner` configuration as follows:

```js
export const runnerConfig: Partial<RunOptions> = {
skipJsErrors: true, // Should ideally be set to false
quarantineMode: { successThreshold: 1, attemptLimit: 3 }, // Tests are only marked as failed when they fail three times
selectorTimeout: 20000,
assertionTimeout: 20000,
pageLoadTimeout: 20000,
speed: 1,
browserInitTimeout: 720000,
developmentMode: true,
};
```

and then share a screenshot of the `Console` panel in the same way as in the issue's description:

![](https://user-images.githubusercontent.com/27851699/216278834-150bb990-4b81-423f-84ba-ceb9b980df8f.png)

The shared screenshot should contain full function names in the call stack.

**Comment by ChristianThingKnudsen** [comment]
There is no change in the console panel. This is what I get after applying the above:
<img width="614" alt="image" src="https://user-images.githubusercontent.com/27851699/223389346-dc5aa89f-77fe-4cdd-b1ed-956c142c7b4e.png">

**Comment by ChristianThingKnudsen** [comment]
I have just shared a simple example to [support@devexpress.com](mailto:support@devexpress.com) to our private test server with the same topic as this issue where you hopefully can reproduce the error.

**Comment by AlexKamaev** [comment]
I managed to reproduce the issue. Thank you for sharing the example.

**Comment by AlexKamaev** [comment]
We cannot give you any precise estimates of when the issue will be resolved. We would like to have further access to your test server if possible.

**Comment by AlexKamaev** [comment]
Hi @moniquelive,

Thank you for sharing this information. We will follow-up with you once we have specific information to share.

**Comment by JordanDalton** [comment]
Ran into a situation like this today this did the trick:

`--experimental-proxyless`

**Comment by ChristianThingKnudsen** [comment]
Unfortunately in my case the flag `--experimental-proxyless` is causing other issues in my provided example. Some https requests are now returning `401 (Unauthorized)` which they did not before. So this is not a viable solution in my case.

**Comment by AlexKamaev** [comment]
> Unfortunately in my case the flag `--experimental-proxyless` is causing other issues in my provided example. Some https requests are now returning `401 (Unauthorized)` which they did not before. So this is not a viable solution in my case.

Could you please create a separate issue for this inquiry?
Please note that in TestCafe `2.5.0` and newer, the `--experimental-proxyless` flag is called `--native-automation`.

**Comment by AlexKamaev** [comment]
Hi @ChristianThingKnudsen,

Thank you for sharing your results. We appreciate it.

**Comment by miherlosev** [comment]
Hi @ChristianThingKnudsen,

I tried to reproduce the issue on the latest commit, and test execution stopped on the failed assertion.
![](https://github.com/DevExpress/testcafe/assets/4133518/42a050ee-0dfd-4c7e-abd6-0c03909009e7)
I guess the logic of the tested application was changed. Could you please update the example and share it with us?

**Comment by ChristianThingKnudsen** [comment]
The following code should work to reproduce the issue:

import { Selector } from "testcafe";

fixture("Child nodes fixture").page("https://oth-test.oth.io/clinician/");

const loginForm = Selector("#login");
const usernameInput = loginForm.find("#username");
const passwordInput = loginForm.find("#password");
const loginButton = loginForm.find("#submit");

const patientOverview = Selector("#list-patient");
const scott = patientOverview.find("a").withText("Scott Malkinson");

const departments = Selector("#list-departments");
const department = departments.find("a").withText("Diabetes Organization");

const messagesBtn = Selector("#patient-messages-menu-button").withText(
  "Messages"
);

test("Should not throw childNodes of undefined", async (t) => {
  await t
    .typeText(usernameInput, "DoctorHartman")
    .typeText(passwordInput, "abcd1234")
    .click(loginButton);

  await t.expect(patientOverview.exists).ok();
  await t.expect(scott.exists).ok();
  await t.click(scott);

  await t.expect(messagesBtn.exists).ok();
  await t.click(messagesBtn);

  await t.expect(department.exists).ok();
  await t.click(department);
  await t.debug();
});

The error was due to the messagesBtn selector having withExactText instead of withText.

Sincerly
Christian Thing Knudsen
________________________________
Fra: Mikhail Losev ***@***.***>
Sendt: 18. maj 2023 04:50
Til: DevExpress/testcafe ***@***.***>
Cc: ChristianThingKnudsen ***@***.***>; Mention ***@***.***>
Emne: Re: [DevExpress/testcafe] Uncaught TypeError: Cannot read properties of undefined (reading 'childNodes') (Issue #7489)


Hi @ChristianThingKnudsen<https://github.com/ChristianThingKnudsen>,

I tried to reproduce the issue on the latest commit, and test execution stopped on the failed assertion.
[https://user-images.githubusercontent.com/4133518/239121318-42a050ee-0dfd-4c7e-abd6-0c03909009e7.png]<https://user-images.githubusercontent.com/4133518/239121318-42a050ee-0dfd-4c7e-abd6-0c03909009e7.png>
I guess the logic of the tested application was changed. Could you please update the example and share it with us?

**Comment by ChristianThingKnudsen** [comment]
I just sent you an email with the updated code. The error above was due to the `messagesBtn` selector having `withExactText` instead of `withText`. If you correct this you should be able to reproduce

**Comment by ChristianThingKnudsen** [comment]
@Aleksey28 Do you know if this is going to be a part of testcafes upcoming version 3.0.0?

**Comment by ChristianThingKnudsen** [comment]
@Aleksey28 I have just tested the newly released Testcafe v.3.0.0. It works locally on my computer, but it seems like the issue still persists on the docker image.

**Comment by Aleksey28** [comment]
@ChristianThingKnudsen please ensure that you use Testcafe v.3.0.0 in the docker image. Execute the `testcafe -v` command before starting tests.

**Comment by ChristianThingKnudsen** [comment]
@Aleksey28  Using your latest docker image or the one with the tag: `3.0.0` It prints this in the console when executing `testcafe -v`:

```
Using locally installed version of TestCafe.
2.6.1
```

I am using bitbucket pipeline and have tried to using: `image: testcafe/testcafe` and `image: testcafe/testcafe:3.0.0`

**Comment by Aleksey28** [comment]
@ChristianThingKnudsen,

I executed the 'docker pull' command for the `testcafe/testcafe` image with the `latest` tag and the `docker run testcafe/testcafe -v` command. As a result, `3.0.0` was printed. Please reinstall the `testcafe/testcafe` image.

**Comment by ChristianThingKnudsen** [comment]
My apologies. The issue was due to Bitbucket caching an old Testcafe version in the pipeline even though I used the correct Docker image.
